src/commands/signzone.rs

```diff
@@ -0,0 +1,285 @@
+        default_value_t = false,
+        requires = "nsec3",
+        conflicts_with = "nsec3_opt_out"
+    )]
+    nsec3_opt_out_flags_only: bool,
+
+    /// Set the opt-out flag on all NSEC3 RRs and skip unsigned delegations
+    #[arg(
+        help_heading = Some("NSEC3 (when using '-n')"),
+        short = 'A', // Matches BIND dnssec-signzone
+        default_value_t = false,
+        requires = "nsec3",
+        conflicts_with = "nsec3_opt_out_flags_only"
+    )]
+    nsec3_opt_out: bool,
+
+    /// Hash only, don't sign
+    #[arg(short = 'H', default_value_t = false)]
+    hash_only: bool,
+
+    /// Use layout in signed zone and print comments on DNSSEC records
+    #[arg(short = 'b', default_value_t = false)]
+    diagnostic_comments: bool,
+
+    /// Origin for the zone (for zonefiles with relative names and no $ORIGIN)
+    #[arg(short = 'o')]
+    origin: Option<Name<Bytes>>,
+
+    /// The zonefile to sign
+    #[arg(value_name = "zonefile")]
+    zonefile_path: PathBuf,
+
+    /// The keys to sign the zone with
+    ///
+    /// keys must be specified by their base name (usually
+    /// K<name>+<alg>+<id>), i.e. WITHOUT the .private extension.
+    #[arg(value_name = "key")]
+    key_paths: Vec<PathBuf>,
+}
+
+impl SignZone {
+    pub fn execute(self) -> Result<(), Error> {
+        let mut records = self.load_zone()?;
+
+        //---
+
+        // Import the specified key.
+        let mut keys = vec![];
+        for key_path in self.key_paths {
+            let old_ext = key_path.extension().unwrap().to_str().unwrap();
+            let new_ext = format!("{}.private", old_ext);
+            let private_key_path = key_path.with_extension(new_ext).display().to_string();
+            let private_data = std::fs::read_to_string(&private_key_path).map_err(|err| {
+                Error::from(format!(
+                    "Unable to load private key from file '{}': {}",
+                    private_key_path, err
+                ))
+            })?;
+
+            let old_ext = key_path.extension().unwrap().to_str().unwrap();
+            let new_ext = format!("{}.key", old_ext);
+            let public_key_path = key_path.with_extension(new_ext).display().to_string();
+            let public_data = std::fs::read_to_string(&public_key_path).map_err(|err| {
+                Error::from(format!(
+                    "Unable to load public key from file '{}': {}",
+                    private_key_path, err
+                ))
+            })?;
+
+            let secret_key =
+                domain::sign::SecretKeyBytes::parse_from_bind(&private_data).map_err(|err| {
+                    Error::from(format!(
+                        "Unable to parse BIND formatted private key file '{}': {}",
+                        private_key_path, err
+                    ))
+                })?;
+
+            let public_key_info: domain::validate::Key<Bytes> =
+                domain::validate::Key::parse_from_bind(&public_data).map_err(|err| {
```

Comment:
It's unfortunate that this is generic over `Octs` right now, but I'd suggest omitting the explicit type for the variable and just writing `domain::validate::Key::<Bytes>::parse_from_bind(...)`, that'll be a bit shorter.